explainer.md

```diff
@@ -0,0 +1,329 @@
+    <x-option id='opt3'>Option 3</x-option>
+ </x-optionlist>
+</x-combobox>
+
+```
+
+
+* Custom UI which uses canvas to draw the UI currently 
+
+A low-level API would bridge these gap,
+allowing authors to bypass artificial limitations or bugs in the platform
+and provide a custom accessible experience where necessary.
+
+### Audience for the proposed API
+
+We don't expect or encourage the average web developer
+to want or need a low-level accessibility API for most things.
+Declarative HTML should be sufficient for most web page authors.
+
+This API is aimed at the relatively small number of developers who create and maintain
+the JavaScript frameworks and widget libraries that power the vast majority of web apps.
+Accessibility is a key goal of most of these frameworks and libraries,
+as they need to be usable in as broad a variety of contexts as possible.
+A low-level API would allow them to work around bugs and limitations
+and provide a clean high-level interface that "just works" for the developers who use their components.
+
+## The Accessibility Object Model
+
+This spec proposes the *Accessibility Object Model*,
+a new API that makes it possible to:
+
+1. explore the accessibility tree automatically generated by the browser from the DOM,
+2. modify the accessibility tree to change what's exposed to assistive technology, and
+3. directly respond to messages from assistive technology.
+
+In the following sections we'll explore each of these major categories of functionality
+and how we propose the accessibility API would work.
+
+### Exploring the accessibility tree
+
+The most basic thing that's possible using the proposed Accessibility Object Model
+is exploring and querying the existing accessibility tree.
+
+With the AOM, every DOM node has a property **`accessibleNode`**
+that accesses that element's corresponding object in the accessibility tree,
+if it has one.
+Most DOM nodes will have an accessible node, with a few exceptions -
+an element might not have an accessible node if it's not currently attached to a visible Document,
+or if it's not currently displayed, for example.
+
+Accessing an element's accessible node allows you to determine its role.
+This example shows how you could figure out the role assigned to an HTML INPUT element with a type of "range":
+
+```html
+<label>
+  Rating:
+  <input id="myinput" type="range" value="5" min="1" max="10">
+</label>
+<script>
+  var input = document.querySelector("#myinput");
+  var axInput = input.accessibleNode;
+  axInput.role;  // returns "slider"
+</script>
+
+```
+
+Most ARIA attributes have a corresponding property on an accessible node.
+In this particular case, we can access
+the min, max, and current value of the slider
+and its text label, among other things.
+
+```js
+axInput.rangeValue;  // returns 5.0
+axInput.rangeMin;    // returns 1.0
+axInput.rangeMax;    // returns 10.0
+axInput.label;       // returns "Rating:"
+```
+
```

Comment:
feels like you're missing sections on how to modify the accessibility tree and respond to actions (items 2 and 3 from above)